File: tzcode/hook/task.py
import frappe

def validate(doc, method):
    sync_with_parent(doc)

def sync_with_parent(doc):
    if doc.is_group:
        filters = {"parent_task": doc.name}
        for name, in frappe.get_list("Task", filters, as_list=True):
            child = frappe.get_doc("Task", name)
            child.update({
                "color": doc.color,
                "project": doc.project,
            })
    else:
        if not doc.parent_task:
            return 
        color, project = frappe.get_value(
            "Task",
            doc.parent_task,
            ["color", "project"]
        )
        doc.update({
            "color": color,
            "project": project,
        })

File: tzcode/public/js/task.js
frappe.ui.form.on("Task", {
    is_group(frm) {
        frm.trigger("req_color");
    },
    color(frm) {
        frm.trigger("req_color");
    },
    req_color(frm) {
        frm.set_df_property("color", "reqd", !!frm.doc.is_group)
    }

})